src/charactersheet/viewmodels/dm/encounter_add_edit_modal/index.js

```diff
@@ -74,11 +74,13 @@ export function EncounterAddEditModalViewModel({ encounter, openModal, onsave })
 
     self._dataHasChanged = () => {
         const encounter = ko.unwrap(self.encounter);
-        if (ko.unwrap(encounter.uuid)) {
+        if (encounter && ko.unwrap(encounter.uuid)) {
```

Comment:
Is the `if` for the case where we're editing the existing encounter and the `else` for adding a new one?

Reply:
Yep exactly.